### ISSUE TITLE
Reproduction results far below paper

### ISSUE THREAD

**Issue by Grace160** [MichaelMaiii/SynBrain#3]
Hi, thank you very much for releasing the code and especially the preprocessing script — it helped me successfully run the training pipeline.

However, I’m seeing a large performance gap between my results and those reported in the paper, even after carefully checking my setup.

I followed the default settings, using hidden_dim=4096, fused_attn=False (also tried True, but with no improvement), and ensured that all other parameters match the release. For voxel inputs, I tried both:

Averaging 3 presentations
Randomly sampling 3 presentations
Unfortunately, both led to quite poor results. I would appreciate your help on the following:

Is the voxel processing in the paper based on averaging or sampling (or something else)?
Is the released preprocessing script complete, or are there any additional steps not included?
Is the current codebase the exact one used in the paper, or could there be missing implementation details?
Below are my results on Subject 1 for reference:

```
Low-Level Metrics:
PixCorr: -0.0270
SSIM: 0.1740
Alex(2): 0.4996
Alex(5): 0.4954

High-Level Metrics:
Incep: 0.5061
CLIP: 0.5063
Eff: 0.9844
SwAV: 0.6072

Retrieval:
fwd_percent_correct: 0.0026
bwd_percent_correct: 0.0022
fwd_percent_correct_recon: 0.0450
bwd_percent_correct_recon: 0.0077
fwd_percent_correct_fm: 0.0039
bwd_percent_correct_fm: 0.0046

Similarity:
MSE: 0.1119, Pearson: 0.5248, CKA: 0.5131, Cosine: 0.6345
```

Thanks again for your great work — any clarification or debugging suggestions would be greatly appreciated!

**Comment by MichaelMaiii** [comment]
Hi, something is definitely wrong with your setup; fwd_percent_correct < 0.1% is never shown in my experiments. Instead, this metric is expected to be above 90% on the test set for BrainVAE. In this case, don't bother with the S2N Mapper stage until you've corrected the BrainVAE stage.

For data processing, you can follow the processing script to generate all fMRI data (3 repetitions) -> [27000, 15724] for the training set of subject 1, and use the dataloader directly.
 
Maybe I'll provide the data file and the checkpoint to reproduce the results. But it won’t be updated until the end of this month, as I’ve been busy with a recent deadline.

**Comment by Grace160** [comment]
Thank you very much for your suggestions regarding data processing methods. After implementing your advice, the results have shown some improvement. However, the performance is still below that reported in the original paper. The current results are as follows:

```
Low-Level Metrics:
PixCorr: 0.0383
SSIM: 0.2600
Alex(2): 0.5121 (2-way percent correct)
Alex(5): 0.5048 (2-way percent correct)

High Level Metrics:
Incep: 0.4893 (2-way percent correct)
CLIP: 0.5066 (2-way percent correct)
Eff: 0.9738
SwAV: 0.6322

Retrieval Metrics:
fwd_percent_correct: 0.9611
bwd_percent_correct: 0.8610
fwd_percent_correct_recon: 0.9874
bwd_percent_correct_recon: 0.7857
fwd_percent_correct_fm: 0.9484
bwd_percent_correct_fm: 0.9814

'MSE': 0.07969923541819056, 'Pearson': 0.7181243427028853, 'CKA': 0.612355649471283, 'Cosine': 0.7693755`
```

All experimental parameters were used as provided in the open-source code, and no modifications were made without justification. I also tried several possible parameter adjustments, but none of them achieved the reported performance.

I’m not sure where the issue might be. When you have time, I would greatly appreciate it if you could help confirm the parameters in the open-source code or share any further suggestions you might have.

Thank you again for your help and support!

**Comment by MichaelMaiii** [comment]
Hi, I wanna know how you get the "all_images.pt" in eval.py, I miss that in my code, if you use that file from MindEye2, that would be wrong cause the image sorting in the file is not the same.

**Comment by MichaelMaiii** [comment]
I have uploaded the [save_images_eval.py](https://github.com/MichaelMaiii/SynBrain/blob/main/data/save_images_eval.py) file to generate "all_images.pt" in this work, you can try again.

**Comment by Grace160** [comment]
Thank you for your reply. I resolved the issue by replacing the “all_images.pt” file obtained from MindEye2 with the one generated using your provided code. I really appreciate your help!

**Comment by Grace160** [comment]
Hello, I’ve noticed that when reproducing the results, sometimes the training produces NaN loss, while other times it trains normally. Have you encountered this issue as well?

**Comment by MichaelMaiii** [comment]
I’ve encountered NaN loss when training with FP16. After switching to FP32 training, the issue no longer occurred.
